src/WebJobs.Script.WebHost/Diagnostics/SystemLogger.cs

```diff
@@ -123,17 +125,29 @@ public void Log<TState>(LogLevel logLevel, EventId eventId, TState state, Except
 
             string invocationId = string.Empty;
             object scopeValue = null;
-            if (scopeProps != null && scopeProps.TryGetValue(ScriptConstants.LogPropertyFunctionInvocationIdKey, out scopeValue) && scopeValue != null)
+            string scopeActivityId = null;
+            if (scopeProps != null)
             {
-                invocationId = scopeValue.ToString();
+                if (scopeProps.TryGetValue(ScriptConstants.LogPropertyFunctionInvocationIdKey, out scopeValue) && scopeValue != null)
+                {
+                    invocationId = scopeValue.ToString();
+                }
+
+                // For Http function invocations we want to stamp invocation logs with
+                // the request ID for easy correlation with incoming Http request logs.
+                if (scopeProps.TryGetValue(ScriptConstants.LoggerHttpRequest, out scopeValue) && scopeValue != null)
```

Comment:
This scope value is added [here](https://github.com/Azure/azure-functions-host/blob/16cf88eca5d144f226b3b89d2007d4011c66dbe5/src/WebJobs.Script.WebHost/Middleware/FunctionInvocationMiddleware.cs#L135) as part of function invocation

Reply:
A bit of a nit, but this would be safe and more concise:

``` csharp
 // For Http function invocations we want to stamp invocation logs with
 // the request ID for easy correlation with incoming Http request logs.
 if (scopeProps.TryGetValue(ScriptConstants.LoggerHttpRequest, out HttpRequest request))
 {
     scopeActivityId = request.GetRequestId();
 }
```

Reply:
Fixed